### DISCUSSION TITLE
[help] how to figure out which row in params is failing in tar_render_rep

### DISCUSSION THREAD

**Discussion by erinsteiner-NOAA** [ropensci/tarchetypes#205] [Help]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

I was wondering if anyone has figured out a clever way to figure out which row in the params data.frame is failing? I have a targets workflow that renders ~200 reports using the tar_render_rep function. When a branch fails to complete, it's really tricky to figure out which line in the params data.frame is the one that failed. I was curious if anyone has figured out how to either name the branches so that they can be matched to the params data.frame row? Or print an iteration number to the the terminal? Or maybe there's another solution I'm not thinking of? Thanks for considering my question!

![image](https://github.com/user-attachments/assets/2a00082e-fdf2-45b3-9aab-6bc3b65f9eb2)

**Comment by ginolhac** [comment]
Hello,

you can add a column in the data.frame with `tar_name()` so the branch hash is included

**Comment by erinsteiner-NOAA** [comment]
@ginolhac - thanks for the tip about tar_name() - I used it to print the current run to a csv that I can monitor in real time with the info I need to know about which record is failing:

![image](https://github.com/user-attachments/assets/c8814e94-d1d7-49ab-9456-5a7ce12f3d15)
